### DISCUSSION TITLE
Discussions tab

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#2365] [Enhancement ideas (and their technical discussions)]
Hi

Can we use the discussions tab of Github soon?

It will allow everyone to discuss important things, GD's vision for the future, features, it can also be a great place for important messages for developers.

I'm thinking about[ this kind of important comment.](https://github.com/4ian/GDevelop/pull/2157#issuecomment-757008907)

A Suggestion features section can be useful to remove suggestions in the Issues tab.
And redirect suggestions from the forum to this new suggestion section in discussions  tab.
What's the reason for this? Because we have three places to make suggestions, the trello, the forum, and github.

Regroup everything in one place allows to have a global vision of the requests, the approved suggestions goes to Trello list.

And this way suggestions and bugs are no longer mixed up in the github Issues. Issues tab will be only for bugs.

A Q&A section about the engine can be useful too!

So we would have sections:
- Feature request (when approved it go on Trello, replace "Careful thinking/design or refactoring needed" label in issues tab) (ex: [1](https://github.com/4ian/GDevelop/issues/2159), [2](https://github.com/4ian/GDevelop/issues/2163))
- Q&A (about the engine, replace the "question" label in issues tab)
- Devs note/devs blog (important update for devs)

I start from the observation that some of the suggestions and bugs in the forum are not always reported on Github because there are not enough people to make the bridge between the two websites.
And even less between the Trello, the forum and Github.
Less is more, isn't it?

**Comment by 4ian** [comment]
I think that's mostly fine, I'm just afraid of extra noise from users that would ask question here rather than on the forum/Discord. We have to be sure that it's clear that:
- Help for creation is on the forum/Discord
- GitHub discussions is only for Q&A related to development.

For "feature requests", I'm still a bit torn apart between keeping them on the forum and then switching to a discussion on GitHub when there is enough technical stuff or when the request make enough sense. I'm also a bit afraid of people jumping the ship and directly opening discussions on GitHub that are just "can we have 3d plz??".

The good part will be to have issues only for bugs, that will be a very good thing :)

**Comment by arthuro555** [comment]
The main issue I have with feature requests on the forum is that they basically drown most of the time. For example, I made a proposition for one of my favorite things, the [ternary operator](https://forum.gdevelop-app.com/t/ternary-operator/29078?u=arthuro555). It got a few likes, so it wasn't a bad idea and maybe could have been taken to GitHub, but no one commented on it to have a discussion on what else to think about before porting it to GitHub, and soon the post drowned and got forgotten. It is the same for other feature, where I am like "Hey we should definitely add that" but then it never gets ported to GitHub and never gets implemented. 

Having a place just for feature request would let them not drown, and if it is github discussion it could be easily ported into GitHub issues if considered a good feature request.

**Comment by 4ian** [comment]
It's partially "by design", in the sense that we'll always get more feature requests that we can handle - even if you and I was working 24/7 on GDevelop (that's also the reason behind the Trello: get a view of all the requests and have votes to show that we need a strong community ask before doing something).
I fear that the same could happen if we move to GitHub without specific rules: we'll get ~5 feature requests per day, and I'll spend 10 minutes for each of them answering or linking to existing answers or saying "well you know why not but are you sure it's a valid request or just something that just you want and what about the added complexity and what about maintenance of it and what about bloating the software instead of keeping a lean core with a set of simple principles?" ;) 

Which amounts to 5x10 = 50 minutes a day to answer to these. As I won't be able to keep up, things will start to "drown" again.

I'm very much keen on opening the discussions on GitHub because it's clearly missing for technical discussions, but for feature requests I'm not 100% sure how to deal with them. In an ideal world, an idea would be submitted, get a few "+1, I totally want this" from 5-10 community members, then it would ring a bell somewhere telling "this feature is requested a lot - it will be added to the roadmap automatically so it can gather more votes, and developed if GDevelop core developers are happy about the design of this feature (that they can discuss on GitHub or on the Trello comments)"
(of course, we could also have much simpler process for very small features/fixes)

Note that it's not a new problem :) The Trello roadmap was opened because people were complaining that they had no visibility on new features, and that their feature requests were not taken into account (which is partially false: I'm reading a lot of things, it's just that not everything can or should be done).
I think it's also due to the organic way we organise things as an open-source project: we can't develop everything, so if someone wants to really push something, they can try to demonstrate interest, gather community support around it, develop it themselves (but getting an approval from myself is better at least for core features).

Other projects like Godot have things like "proposals": https://godotengine.org/article/introducing-godot-proposals-repository - maybe something we learn from.

**Comment by tristanbob** [comment]
I do like the idea of having one (or two) places for each specific purpose:

- Requesting HELP on how to use GDevelop
  - Real-time format for quick response on "simple" issues = Discord
  - Permanent record (Google searches often land here) that is best for complex questions = Forums
- Documentation for users = Wiki
- Reporting BUGS with GDevelop = Github Issues: "Report bug"
- Suggesting new features/improvements = Github Issues: "Suggest feature"
- Technical and strategic talks about how GDev should be built = Discussions
- Planning and prioritizing tasks = Trello


Perhaps there is a possible way to get the best of both worlds (using GitHub to track more things) without the core-dev team spending time on every request?  

Is it possible for a team of moderators to tag issues that deserve review by the core team?

**Comment by INNOVATIVEGAMER** [comment]
@4ian Is whole GD is converting into TypeScript ? if so Is there any plans which files are getting converted to TS ? or contributors are randomly changing files to TS ? 
As react also supports TS now. In future is there any chances that IDE and components will convert into TS ?

**Comment by arthuro555** [comment]
> Is whole GD is converting into TypeScript ? if so Is there any plans which files are getting converted to TS ? or contributors randomly changing files to TS ?

The conversion has already happened, and was done mainly automatically via tools. You can find the PR here: #2157 .

> As react also supports TS now. In future is there any chances that IDE and components will convert into TS ?

Personally I'd like a conversion to typescript as well, but we already have flow there and I don't think that there will be a change to typescript anytime soon.


Also, this is unrelated to this issue. You should ask this on the forum, discord or trello, or seek the answers on those platforms as there are probably answers there. See [this trello card](https://trello.com/c/qy3QQ7qD/350-convert-the-game-engines-and-extensions-to-typescript) that shows through the checklist what was done and it is in the "Done" list, so there won't be any more work on this. For the second part, see the comments [on this card](https://trello.com/c/O2bFnmS9).

**Comment by 4ian** [comment]
@Bouh @arthuro555 @blurymind looks like we have a discussion tab now 😄 I've moved the questions/enhancement issues there.

Please let's be sure to keep only technical discussion here, the community using GD should prefer using the forum or discord as usual :)

**Reply by 4ian** [reply]
Things to discover:
* How we mark things as "Good First Issues" now for enhancements?
* can we close discussions? Or not?

**Reply by arthuro555** [reply]
Should experienced contributors leave the "Good First Issues" to new contributors?

**Reply by Bouh** [reply]
We can lock discussions
![image](https://user-images.githubusercontent.com/1670670/108636101-7f499a80-7483-11eb-9872-97f24d7194de.png)

**Reply by 4ian** [reply]
> Should expereienced contributors leave the "Good First Issues" to new contributors?

Not necessarily, please go ahead if you can address it!
